sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/MSITokenTests.java

```diff
@@ -3,34 +3,107 @@
+
     @Test
     public void canParseLong() {
-        MSIToken token = new MSIToken("fake_token", "1578668608", null);
-        MSIToken token2 = new MSIToken("fake_token", null, "3599");
-        MSIToken token3 = new MSIToken("fake_token", "1578668608", "3599");
+        MSIToken token = new MSIToken("fake_token", "1578668608", null, null);
+        MSIToken token2 = new MSIToken("fake_token", null, "3599", null);
+        MSIToken token3 = new MSIToken("fake_token", "1578668608", "3599", null);
 
         Assert.assertEquals(expected.toEpochSecond(), token.getExpiresAt().toEpochSecond());
         Assert.assertTrue((token2.getExpiresAt().toEpochSecond() - OffsetDateTime.now().toEpochSecond()) > 3500);
         Assert.assertEquals(expected.toEpochSecond(), token3.getExpiresAt().toEpochSecond());
     }
 
+    @Test
+    public void canParseRefreshes() {
+        try (MockedStatic<OffsetDateTime> offsetDateTimeMockedStatic = mockStatic(OffsetDateTime.class, CALLS_REAL_METHODS)) {
+
+            offsetDateTimeMockedStatic.when(() -> OffsetDateTime.now((ZoneId) any())).thenReturn(expected);
+            offsetDateTimeMockedStatic.when(() -> OffsetDateTime.now((Clock) any())).thenReturn(expected);
+            offsetDateTimeMockedStatic.when(OffsetDateTime::now).thenReturn(expected);
+
+            OffsetDateTime now = OffsetDateTime.now();
+            OffsetDateTime expiration = now.plusHours(12);
+            OffsetDateTime expirationMinusOneHour = expiration.minusHours(1);
+            Duration expirationMinusOneHourSeconds = Duration.between(expirationMinusOneHour, expected);
+            OffsetDateTime expirationMinusElevenHours = expiration.minusHours(11);
+            Duration expirationMinusElevenHoursSeconds = Duration.between(expirationMinusElevenHours, expected);
+
+            MSIToken expirationMinusOneHourToken = new MSIToken("fake_token",
+                String.valueOf(expirationMinusOneHour.toEpochSecond()),
+                String.valueOf(expirationMinusOneHourSeconds.getSeconds()), null);
+
+            MSIToken expirationMinus11HoursToken = new MSIToken("fake_token",
+                String.valueOf(expirationMinusElevenHours.toEpochSecond()),
+                String.valueOf(expirationMinusElevenHoursSeconds.getSeconds()), null);
+
+            MSIToken hasRefresh = new MSIToken("fake_token",
+                String.valueOf(expirationMinusElevenHours.toEpochSecond()),
+                String.valueOf(expirationMinusElevenHoursSeconds.getSeconds()),
+                String.valueOf(240));
+
+            Assert.assertEquals(240, hasRefresh.getRefreshIn());
+            Assert.assertEquals(Duration.ofHours(1).getSeconds(), expirationMinus11HoursToken.getRefreshIn());
+            Assert.assertEquals(Duration.between(now, expirationMinusOneHour).getSeconds() / 2, expirationMinusOneHourToken.getRefreshIn());
+
+        }
+
+    }
+
+    @Test
+    public void canDeserialize() {
```

Comment:
Would be good to add a canDeserialize case without refresh_in value to ensure backward compatibility.